### ISSUE TITLE
Unable to find $(DevEnvDir)\Extensions\Xamarin.VisualStudio/*.dlls 

### ISSUE THREAD

**Issue by CartBlanche** [actions/virtual-environments#4566]
### Description

I'm trying to build some Xamarin related projects using Github Actions.
I looked here and but couldn't find what dlls are installed with each extension....
https://github.com/actions/virtual-environments/blob/main/images/win/Windows2022-Readme.md#visual-studio-enterprise-2022

The actual path that seems to be missing on the VMs is 
`$(DevEnvDir)\Extensions\Xamarin.VisualStudio\Mono.Debugging.*.dll`
and
`$(DevEnvDir)\Extensions\Xamarin.VisualStudio\Mono.Debugger.Soft.dll`

Which are referenced in my csproj and works fine from the command line here on my machine, for both VS2019 and VS2022 with MSBuild, but it doesn't find those in the VM. 

A log for failing build can be see here - https://github.com/WildernessLabs/VS_Win_Meadow_Extension/runs/4285115410?check_suite_focus=true

Thanks!

### Virtual environments affected

- [ ] Ubuntu 18.04
- [ ] Ubuntu 20.04
- [ ] macOS 10.15
- [ ] macOS 11
- [X] Windows Server 2016
- [X] Windows Server 2019
- [X] Windows Server 2022

### Image version and build link

The PR in question, where I'm trying to get CI working is here - 
https://github.com/WildernessLabs/VS_Win_Meadow_Extension/pull/35

### Is it regression?

No

### Expected behavior

If the correct Xamarin SDKs are installed, then it should just build

### Actual behavior

It cannot find the aforementioned DLLs in the Extensions sub-dir.

### Repro steps

As per the build link above

**Comment by dibir-magomedsaygitov** [comment]
Hello @CartBlanche. Thank you for your report. We will take a look.

**Comment by vsafonkin** [comment]
Hi @CartBlanche, probably the thing is Visual Studio extensions are installed with admin permissions on the Windows runner. According [Visual Studio docs](https://github.com/MicrosoftDocs/visualstudio-docs/blob/master/docs/ide/finding-and-using-visual-studio-extensions.md#per-user-and-administrative-extensions), location of these extentions is `<Visual Studio installation folder>\Common7\IDE\Extensions` directory. I've checked it for `windows-latest` (Visual Studio 2019) and found `Xamarin.VisualStudio` extentions in this folder:
```
Get-ChildItem "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\Extensions\Xamarin.VisualStudio"
```
<img width="927" alt="Screenshot 2021-11-25 at 1 20 49 PM" src="https://user-images.githubusercontent.com/7628945/143423728-25cde73f-b20d-43ce-b404-f743f0514489.png">

**Comment by CartBlanche** [comment]
@vsafonkin Thanks for the reply. So what do I need to do for my VS extension to see this path on the runner? 

I know that locally, on Windows, assuming I have the .NET Mobile workload installed, my VS Extension sees that path and builds perfectly from both the command line and the IDE. So I sort of expect it to "Just Work"(tm) on the runner too, assuming everything there is installed identically to what I have locally. 

So is there anything else I need to do to get this working in my CI?

**Comment by CartBlanche** [comment]
@dibir-magomedsaygitov, @MaksimZhukov Do either of you have any other suggestions, as I can't merge my PR until my build goes green??

**Comment by vsafonkin** [comment]
@CartBlanche, we are still investigating it and try to find properly decision for you, thank you for your patience!

**Comment by vsafonkin** [comment]
@CartBlanche, looks like `DevEnvDir` macros is not defined on the Windows runner. Could you try this step in your workflow to build your project?
```yaml
- name: Build VS2019 Extension
  id: VS2019-Extension
  run: msbuild main/VS_Meadow_Extension.2019.sln -t:rebuild /p:Configuration=Release
  env:
    DevEnvDir: 'C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE'
```

**Comment by CartBlanche** [comment]
@vsafonkin Yes creating our own DevEnvDir environment variable helps and allows the VS2019 project to build. So thanks for that!!

I now have an issue with VS2022 (is that installed in the runner anywhere?), so wondering if I should raise another issue for that or continue?

**Comment by vsafonkin** [comment]
Closed, @CartBlanche please contact us if you have any questions.
